src/bpf-loader.js

```diff
@@ -0,0 +1,72 @@
+// @flow
+
+import * as fs from 'file-system';
+import * as elfy from 'elfy';
+
+import { Account, PublicKey, Loader, SystemProgram } from '.';
+import { sendAndConfirmTransaction } from './util/send-and-confirm-transaction';
+import type { Connection } from '.';
+
+/**
+ * Factory class for transactions to interact with a program loader
+ */
+export class BpfLoader {
+  /**
+   * Public key that identifies the NativeLoader
+   */
+  static get programId(): PublicKey {
+    return new PublicKey('0x0606060606060606060606060606060606060606060606060606060606060606');
+  }
+
+  // static parse(programName: string) {
+
+  //   var file = fs.readFileSync(programName);
+  //   var elf = elfy.parse(file);
+  //   var section = elf.body.sections.find(function (sect) {
+  //     return sect.name == '.text.entrypoint';
+  //   });
+  //   // console.log('cwd: ' + process.cwd());
+  //   // console.log('elf: ' + JSON.stringify(elf));
+  //   console.log('section: ' + JSON.stringify(section.data));
+  // }
+
+  /**
+   * Loads a BPF program
+   *
+   * @param connection The connection to use
+   * @param owner User account to load the program with
+   * @param programName Name of the BPF program
+   */
+  static async load(
+    connection: Connection,
+    owner: Account,
+    programName: string,
+  ): Promise<PublicKey> {
+    const programAccount = new Account();
+
+    var file = fs.readFileSync(programName);
```

Comment:
never use `var` (except for if you *really* mean it, and you very rarely do).  Use `const` by default, or `let` for mutable values (yeah `let` is immutable in Rust, you're welcome!)

Reply:
Instead of `fs.readFileSync`:

```js
import fs from 'mz/fs';
const data = await fs.readFile(programName);
```

Reply:
/Users/jack/sandbox/git/solana-web3.js/src/index.js
  3:25  error  Parse errors in imported module './bpf-loader': Line 23: await is a reserved word

  21 |   static parse(programName: string) {
  22 | 
> 23 |     const data = await fs.readFile(programName);

Reply:
`static async parse(programName: string) ...`

functions that use `await` need to be declared as `async` or you get that terrible error message